engine/src/test/java/org/camunda/bpm/engine/test/jobexecutor/JobExecutorStartTest.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
+ * under one or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information regarding copyright
+ * ownership. Camunda licenses this file to you under the Apache License,
+ * Version 2.0; you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.camunda.bpm.engine.test.jobexecutor;
+
+import org.camunda.bpm.engine.impl.interceptor.Command;
+import org.camunda.bpm.engine.impl.jobexecutor.AcquireJobsCommandFactory;
+import org.camunda.bpm.engine.impl.jobexecutor.AcquiredJobs;
+import org.camunda.bpm.engine.impl.jobexecutor.DefaultJobExecutor;
+import org.camunda.bpm.engine.impl.jobexecutor.JobExecutor;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class JobExecutorStartTest {
+
+  private JobExecutor jobExecutor;
+
+  @Before
+  public void setUp() throws Exception {
+    jobExecutor = new DefaultJobExecutor();
+  }
+
+  @After
+  public void tearDown() throws Exception {
+    jobExecutor.shutdown();
+  }
+
+  @Test
+  public void testDefaultInitialization() {
+    jobExecutor.start();
+    assertThat(jobExecutor.getAcquireJobsCmdFactory()).isNotNull();
+    assertThat(jobExecutor.getAcquireJobsRunnable()).isNotNull();
+    assertThat(jobExecutor.isActive()).isTrue();
```

Comment:
The test should have the following structure, and you can assert the `AcquireJobsCommandFactory` class:

```suggestion
    // when
    jobExecutor.start();
    
    // then
    assertThat(jobExecutor.getAcquireJobsCmdFactory()).isNotNull();
    assertThat(jobExecutor.getAcquireJobsRunnable()).isNotNull();
    assertThat(jobExecutor.isActive()).isTrue(); 
    assertThat(jobExecutor.getAcquireJobsCmdFactory()).isInstanceOf(DefaultAcquireJobsCommandFactory.class);
```

Reply:
Sorry, total oversight on my part.

Reply:
> it couples the test a bit too much to the implementation

By this, do you mean the `DefaultAcquireJobsCommandFactory` implementation?
If yes, I think this is the point. This test covers the default initialization use-case, and that one uses the `DefaultAcquireJobsCommandFactory`. It is okay to couple it with this implementation, as we definitely would want to know if the default behavior/implementation ever changes. If in the future, that change is intentional, we can then adjust this test.

Just checking if it `.isNotNull()`  is valid for all cases. It's okay if we leave it like this, as then it would cover the general behavior for this field, i.e. it must not be `null`. But then we should have a third test that asserts the default behavior (if it doesn't already exist).

Reply:
Okay, I'll include it. My original concern was, that the public interface of JobExecuter never guarantees the type of this field to be DefaultAcquireJobsCommandFactory only AcquireJobsCommandFactory. But if we want to assert that it is this default implementation that is selected by default 😏, the assert makes sense. 👍